### PR TITLE
feat: added vep to the somatic variant annotation tools

### PR DESCRIPTION
**Pull request by ericblanc20**
BREAKING CHANGE: most of the previous somatic variant annotation configuration is now put under the jannovar tool

### CODE REVIEW

**Comment by ericblanc20** [comment]
Wait a sec, the resources allocated to the wrapper are not large enough.
I am using a more comprehensive list of samples in order to understand how I should increase it.
Only the resources will be updated in the forthcoming commit, you can start looking at the code at your convenience

**Comment by ericblanc20** [comment]
Based on my initial computation, there 4 minutes of starting time (loading the database & the genome sequence), and 6.5 seconds per variant. So that is 540 variants/hour, or about 13000 variants per day.

For the memory, I see that 16GB is not enough for 1200 variants. I am trying now with 64GB, I hope it will accept about 6000 variants.

**Comment by ericblanc20** [comment]
I agree that the speed & memory requirements are way too high (64GB buys only 1300 variants). I don't know what is the problem, but for a limited number of variants, the wrapper works fine.

I'll try to play with the different parameters, to see if I can reduce the resources needed.

File: snappy_wrappers/wrappers/vep/environment.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+channels:
+    - conda-forge
+    - bioconda
+dependencies:
+    - python
+    - ensembl-vep=102
```
[the remark this answers is not in the document]
Comment (ericblanc20) [reply snappy_wrappers/wrappers/vep/environment.yaml:6]: This is used by GDC data release 32 (for mRNA). The GDC annotation for DNA is older, and doesn't exist on Bioconda. So I picked this version.

File: snappy_pipeline/workflows/somatic_variant_annotation/__init__.py [hunk 2]
```diff
@@ -73,54 +74,59 @@
...
+      max_status_checks_per_second: 10   # throttling of status checks
+      ignore_chroms:            # patterns of chromosome names to ignore
+      - NC_007605  # herpes virus
+      - hs37d5     # GRCh37 decoy
+      - chrEBV     # Eppstein-Barr Virus
+      - 'GL*'      # problematic unplaced loci
+      - '*_decoy'  # decoy contig
+      - 'HLA-*'    # HLA genes
+      use_advanced_ped_filters: false  # whether or not to use the advanced pedigree filters flag
+      path_jannovar_ser: REQUIRED                # REQUIRED
```
[the remark this answers is not in the document]
Comment (ericblanc20) [reply snappy_pipeline/workflows/somatic_variant_annotation/__init__.py:104]: Good point, done

File: snappy_pipeline/workflows/somatic_variant_annotation/__init__.py [hunk 3]
```diff
@@ -263,18 +331,26 @@ def get_result_files(self):
 
         We will process all primary DNA libraries and perform joint calling within pedigrees
```
[the remark this answers is not in the document]
Comment (ericblanc20) [reply snappy_pipeline/workflows/somatic_variant_annotation/__init__.py:332]: It is. Removed...

**Comment by ericblanc20** [comment]
@mbenary The code is now complete, it has run on all the 60 samples from the PDO dataset. Please have a last look, iff you're happy you can merge.

**Review by mbenary — APPROVED** [review]
Looks good.

**Pull request merged**